app/src/main/java/com/cornellappdev/resell/android/ui/screens/main/ChatScreen.kt

```diff
@@ -0,0 +1,224 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.zIndex
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.cornellappdev.resell.android.R
+import com.cornellappdev.resell.android.model.Chat
+import com.cornellappdev.resell.android.ui.components.global.messages.ChatTag
+import com.cornellappdev.resell.android.ui.theme.Secondary
+import com.cornellappdev.resell.android.ui.theme.Style
+import com.cornellappdev.resell.android.ui.theme.Wash
+import com.cornellappdev.resell.android.util.clickableNoIndication
+import com.cornellappdev.resell.android.util.richieMessages
+import com.cornellappdev.resell.android.viewmodel.main.ChatViewModel
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun ChatScreen(
+    messagesViewModel: ChatViewModel = hiltViewModel()
+) {
+    val chatUiState = messagesViewModel.collectUiStateValue()
+    var text by remember { mutableStateOf("") }
+    val insets = WindowInsets.isImeVisible
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(bottom = if (insets) 24.dp else 0.dp, top=if (insets) 304.dp else 0.dp),
+    ) {
+        ChatHeader (
+            chat = chatUiState.currentChat ?: Chat(chatId = -1) ,
+            onBackPressed = { messagesViewModel.onBackPressed() },
+            modifier = Modifier
+                .align(Alignment.TopCenter)
+                .background(Color.White)
+                .zIndex(1f)
+        )
+        LazyColumn (
+            modifier = Modifier
+                .fillMaxWidth()
+                .fillMaxHeight()
+                .padding(12.dp)
+                .padding(end = 8.dp)
+        ) {
+            item { Spacer(modifier = Modifier.height(128.dp)) }
+            item {
+                ChatMessage(
+                    imageUrl = "https://media.licdn.com/dms/image/D4E03AQGOCNNbxGtcjw/profile-displayphoto-shrink_200_200/0/1704329714345?e=2147483647&v=beta&t=Kq7ex1pKyiifjOpuNIojeZ8f4dXjEAsNSpkJDXBwjxc",
+                    messages = richieMessages(5),
+                    messageSender = { str, funct, i ->
+                        OtherMessage(str, funct, i ?: 0)
+                    }
+                )
+            }
+            item {Spacer(modifier = Modifier.height(12.dp))}
+            item {
+                ChatMessage(
+                    imageUrl = null,
+                    messageSender = { str, funct, i ->
+                        UserMessage(funct)
+                    },
+                    messages = richieMessages(5)
+                )
+            }
+            item {
+                Spacer(modifier = Modifier.height(128.dp))
+            }
+        }
+        Column(modifier = Modifier.align(Alignment.BottomCenter)){
+            Row(
```

Comment:
Plz format file :-)

Reply:
Also, let's pull out this `Column` and its contents into its own private Composable for readability. This seems to be the bottom screen content? You can call it `ChatFooter`.